File: Backend/Springboot/payment-service/src/main/java/com/beverage/payment/domain/model/Payment.java
package com.beverage.payment.domain.model;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.math.BigDecimal;
import java.time.Instant;
import java.util.UUID;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Payment {
    private UUID id;
    private UUID orderId;
    private String orderCode;
    private UUID userId;
    private BigDecimal amount;
    private PaymentMethod paymentMethod;
    private PaymentStatus status;
    private String transactionId;
    private String paymentUrl;
    private String gatewayResponse;
    private Instant paidAt;
    private String idempotencyKey;
    private Instant expiredAt;
    private Instant createdAt;
    private Instant updatedAt;

    public boolean isExpired() {
        return status == PaymentStatus.PENDING && expiredAt != null && expiredAt.isBefore(Instant.now());
    }
}

File: Frontend/admin/src/types/payment.ts
export type PaymentStatus = 'PENDING' | 'PROCESSING' | 'SUCCESS' | 'FAILED' | 'EXPIRED' | 'REFUNDED' | 'PAID_BY_SHIPPER';
export type RefundStatus = 'PENDING' | 'PROCESSING' | 'COMPLETED' | 'FAILED';
export type PaymentMethod = 'COD' | 'VNPAY';

export interface PaymentDetailResponse {
  id: string;
  orderId: string;
  orderCode: string;
  userId?: string;
  amount: number;
  paymentMethod: PaymentMethod;
  status: PaymentStatus;
  transactionId?: string;
  paymentUrl?: string;
  gatewayResponse?: string;
  paidAt?: string;
  expiredAt?: string;
  orderStatus?: string;
  refundedAmount?: number;
  createdAt: string;
  updatedAt: string;
}

export interface Refund {
  id: string;
  paymentId: string;
  orderId: string;
  orderCode?: string;
  userId?: string;
  amount: number;
  reason: string;
  status: RefundStatus;
  transactionId?: string;
  requestedBy?: string;
  processedAt?: string;
  recipientType?: string;
  shipperName?: string;
  shipperPhone?: string;
  createdAt: string;
  updatedAt: string;
}

export interface RefundCreateRequest {
  amount: number;
  reason: string;
  recipientType?: 'CUSTOMER' | 'SHIPPER';
  shipperName?: string;
  shipperPhone?: string;
}
